contracts/pricers/ChainlinkPricer.sol

```diff
@@ -52,26 +52,34 @@ contract ChainLinkPricer is OpynPricerInterface {
         aggregatorDecimals = uint256(aggregator.decimals());
     }
 
-    /**
-     * @notice modifier to check if sender address is equal to bot address
-     */
-    modifier onlyBot() {
-        require(msg.sender == bot, "ChainLinkPricer: unauthorized sender");
-
-        _;
-    }
-
     /**
      * @notice set the expiry price in the oracle, can only be called by Bot address
      * @dev a roundId must be provided to confirm price validity, which is the first Chainlink price provided after the expiryTimestamp
      * @param _expiryTimestamp expiry to set a price for
      * @param _roundId the first roundId after expiryTimestamp
      */
-    function setExpiryPriceInOracle(uint256 _expiryTimestamp, uint80 _roundId) external onlyBot {
+    function setExpiryPriceInOracle(uint256 _expiryTimestamp, uint80 _roundId) external {
         (, int256 price, , uint256 roundTimestamp, ) = aggregator.getRoundData(_roundId);
 
         require(_expiryTimestamp <= roundTimestamp, "ChainLinkPricer: invalid roundId");
 
+        if (msg.sender != bot) {
+            bool isCorrectRoundId;
+            uint80 previousRoundId = uint80(uint256(_roundId).sub(1));
```

Comment:
I think we  can just use roundId - 1, not 2 extra casting.
If it overflows, it's a big roundId that Chainlink won't return anything, and there's no way in get back to a realistic roundId with the gas limit, from `uint80(-1)`

Reply:
I agree, I also think we will not have `_roundId = 0` where `- 1` will underflow